mender-update/daemon/state_test.cpp

```diff
@@ -2537,11 +2650,20 @@ vector<StateTransitionsTestCase> GenerateStateTransitionsTestCases() {
 			.case_name = "Non_writable_database_in_ArtifactInstall",
 			.state_chain =
 				{
+					"Download_Enter_00",
 					"Download",
+					"Download_Leave_00",
+					"ArtifactInstall_Error_00",
+					"ArtifactRollback_Enter_00",
 					"ArtifactRollback",
+					"ArtifactRollback_Leave_00",
+					// "ArtifactRollbackReboot_Enter_00", // Fails, so not ran
```

Comment:
Hmm, I didn't understand this. Why?

Reply:
Hmm, this is not correct I think. It should still run if it's able, and it's on the error path. After all, `ArtifactFailure_Enter_00` runs, and it's also a SaveState.

Reply:
Then we have to change the `SaveState` functionality.

Reply:
Probably wasn't a save state as I made the test